Code/Framework/AzCore/AzCore/Task/TaskGraphSystemComponent.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Task/TaskGraphSystemComponent.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+
+// Create a cvar as a central location for experimentation with switching from the Job system to TaskGraph system.
+AZ_CVAR(bool, cl_useTaskGraph, false, nullptr, AZ::ConsoleFunctorFlags::Null, "Flag for use of TaskGraph (Note does not disable task graph system)");
+
+namespace AZ
+{
+    void TaskGraphSystemComponent::Activate()
+    {
+        AZ_Assert(m_taskExecutor == nullptr, "Error multiple activation of the TaskGraphSystemComponent");
+
+        if (Interface<UseTaskGraphInterface>::Get() == nullptr)
+        {
+            Interface<UseTaskGraphInterface>::Register(this);
+            m_taskExecutor = aznew TaskExecutor();
+            m_taskExecutor->SetInstance(m_taskExecutor);
+        }
+    }
+
+    void TaskGraphSystemComponent::Deactivate()
+    {
+        if (&TaskExecutor::Instance() == m_taskExecutor) // check that our instance is the global instance (not always true in unit tests)
+        {
+            m_taskExecutor->SetInstance(nullptr);
+        }
+        if (m_taskExecutor)
+        {
+            azdestroy(m_taskExecutor);
+            m_taskExecutor = nullptr;
+        }
+        if (Interface<UseTaskGraphInterface>::Get() == this)
+        {
+            Interface<UseTaskGraphInterface>::Unregister(this);
+        }
+    }
+
+    void TaskGraphSystemComponent::GetProvidedServices(ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("TaskExecutorService"));
```

Comment:
minor, but would suggest a `static constexpr` variable that just computes this once and reuses it (prevents visual studio from boiling the ocean ever so slightly, but also avoids hardcoding the string in more than one place)

Reply:
For the AZ_CRC_CE? As far as I know this is the idiomatic form for O3DE.

We used to have AZ_CRC that would modify the file to hard code the hash, but that has been deprecated since we moved to cmake.

Reply:
As in, `static uint32_t TaskExecutorService::ServiceCRC = AZ_CRC_CE("TaskExecutorService"))` and reusing the `ServiceCRC` variable

Reply:
Updated to use the shared variable.